.github/workflows/unit-tests.yaml

```
@@ -27,3 +27,26 @@ jobs:
           testing-type: units
+
+  test-unit-downstream:
+    runs-on: ubuntu-22.04
+    timeout-minutes: 20
+    needs: test-unit-upstream  # Ensure upstream tests pass before running downstream tests
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
```

actionlint (1.7.4)

the runner of "actions/setup-python@v4" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)